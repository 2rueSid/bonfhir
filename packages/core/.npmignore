.turbo/
fixtures/
src/
templates/
.eslintrc.cjs
jest.config.mjs
jest.setup.mjs
rollup.config.mjs
tsconfig.json
