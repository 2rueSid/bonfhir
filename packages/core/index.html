<!DOCTYPE html>
<html>
  <head>
    <script type="text/javascript" src="dist/r4b/global/index.js"></script>
    <script type="text/javascript">
      console.log(bonfhir.build("Patient", {}));
    </script>
  </head>
  <body>
    <main>
      <h1>Test browser import</h1>
      <p>This file solely exists to be able to test the browser import build (global.ts).</p>
      <p>Just run "npx -y http-server ." after a build and open in a browser, then open the console.</p>
    </main>
  </body>
</html>
