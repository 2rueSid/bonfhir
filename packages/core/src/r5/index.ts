export * from "./builders";
export * from "./bundle-executor";
export * from "./bundle-navigator";
export * from "./date-time-helpers";
export * from "./dedup-search";
export * from "./extensions";
export * from "./fetch-fhir-client";
export * from "./fhir-client";
export * from "./fhir-types.codegen";
export * from "./formatters";
export * from "./lang-utils";
export * from "./merge";
export * from "./mergers/index";
export * from "./narratives.codegen";
export * from "./operations.codegen";
export * from "./patch";
export * from "./patch.codegen";
export * from "./references.codegen";
export * from "./search";
export * from "./search.codegen";
export * as valueFormatters from "./value-formatters/index";
