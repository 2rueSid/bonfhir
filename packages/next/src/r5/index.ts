export * from "./hooks/index.js";
export * from "./subscriptions/index.js";
