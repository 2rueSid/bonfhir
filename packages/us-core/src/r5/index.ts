export * from "./direct-email";
export * from "./jurisdiction";
export * from "./patient";
export * from "./questionnaire-response";
