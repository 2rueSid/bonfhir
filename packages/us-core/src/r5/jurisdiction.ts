/**
 * https://www.hl7.org/fhir/us/core/StructureDefinition-us-core-jurisdiction.html
 */
export const US_CORE_JURISDICTION_URL =
  "http://hl7.org/fhir/us/core/StructureDefinition/us-core-jurisdiction";
