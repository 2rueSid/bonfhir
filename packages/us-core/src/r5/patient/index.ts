export * from "./birth-sex";
export * from "./ethnicity";
export * from "./gender-identity";
export * from "./patient";
export * from "./race";
export * from "./sex";
export * from "./tribal-affiliation";
