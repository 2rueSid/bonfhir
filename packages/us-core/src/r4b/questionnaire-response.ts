/* eslint-disable @typescript-eslint/no-explicit-any */
import { QuestionnaireResponse, SpecialExtension } from "@bonfhir/core/r4b";

/**
 * https://www.hl7.org/fhir/us/core/StructureDefinition-us-core-extension-questionnaire-uri.html
 */
export const US_CORE_QUESTIONNAIRE_URI_URL =
  "http://hl7.org/fhir/us/core/StructureDefinition/us-core-extension-questionnaire-uri";

/**
 * https://hl7.org/fhir/us/core/StructureDefinition-us-core-race.html
 */
export function usCoreQuestionnaireResponseQuestionnaireUri():
  | string
  | undefined {
  return {
    __isSpecialExtension: true,
    __get(target: QuestionnaireResponse): string | undefined {
      const extensionValue = (target._questionnaire?.extension || []).find(
        (x) => (x as any).url === US_CORE_QUESTIONNAIRE_URI_URL,
      );

      return extensionValue?.valueUri;
    },
    __set(
      target: QuestionnaireResponse,
      value: string | undefined,
    ): string | undefined {
      if (!value) {
        if (!target._questionnaire) {
          return;
        }

        target._questionnaire.extension =
          target._questionnaire?.extension?.filter(
            (x) => x.url !== US_CORE_QUESTIONNAIRE_URI_URL,
          );

        if (target._questionnaire.extension?.length === 0) {
          target._questionnaire.extension = undefined;

          if (JSON.stringify(target._questionnaire) === "{}") {
            target._questionnaire = undefined;
          }
        }
        return;
      }

      const newExtensionValue = {
        url: US_CORE_QUESTIONNAIRE_URI_URL,
        valueUri: value,
      };

      if (!target._questionnaire) {
        target._questionnaire = {};
      }

      target._questionnaire.extension = [
        ...(target._questionnaire.extension?.filter(
          (x) => x.url !== US_CORE_QUESTIONNAIRE_URI_URL,
        ) || []),
        newExtensionValue,
      ];

      return value;
    },
  } satisfies SpecialExtension as any;
}

export function usCoreQuestionnaireResponse() {
  return {
    questionnaireUri: usCoreQuestionnaireResponseQuestionnaireUri(),
  };
}
