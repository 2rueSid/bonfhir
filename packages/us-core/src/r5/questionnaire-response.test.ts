import { extendResource } from "@bonfhir/core/r5";
import { usCoreQuestionnaireResponse } from "./questionnaire-response";

describe("questionnaire-response", () => {
  const USCoreQuestionnaireResponse = extendResource("QuestionnaireResponse", {
    ...usCoreQuestionnaireResponse(),
  });

  it("should work", () => {
    const questionnaireResponse = new USCoreQuestionnaireResponse();
    questionnaireResponse.questionnaireUri = "http://example.com";
    expect(questionnaireResponse.questionnaireUri).toBe("http://example.com");

    expect(JSON.stringify(questionnaireResponse)).toMatchSnapshot();

    questionnaireResponse.questionnaireUri = undefined;
    expect(questionnaireResponse.questionnaireUri).toBeUndefined();

    expect(JSON.stringify(questionnaireResponse)).toMatchSnapshot();
  });
});
