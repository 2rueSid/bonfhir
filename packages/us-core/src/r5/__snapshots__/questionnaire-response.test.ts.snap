// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`questionnaire-response should work 1`] = `"{"resourceType":"QuestionnaireResponse","_questionnaire":{"extension":[{"url":"http://hl7.org/fhir/us/core/StructureDefinition/us-core-extension-questionnaire-uri","valueUri":"http://example.com"}]},"text":{"status":"generated","div":"<div xmlns=\\"http://www.w3.org/1999/xhtml\\"><ul></ul></div>"}}"`;

exports[`questionnaire-response should work 2`] = `"{"resourceType":"QuestionnaireResponse","text":{"status":"generated","div":"<div xmlns=\\"http://www.w3.org/1999/xhtml\\"><ul></ul></div>"}}"`;
